xds/internal/balancer/cdsbalancer/cluster_handler_test.go

```diff
@@ -18,1078 +18,650 @@ package cdsbalancer
 	}
-
-	// Invoke a callback for the new logicalDNSService2 - this will fill out the
-	// tree with successful updates.
-	fakeClient.InvokeWatchClusterCallback(xdsresource.ClusterUpdate{
-		ClusterType: xdsresource.ClusterTypeLogicalDNS,
-		ClusterName: logicalDNSService2,
-	}, nil)
-
-	// Behavior: This update make every node in the tree of cluster have
-	// received an update. Thus, at the end of this callback, when you ping the
-	// clusterHandler to try and construct an update, the update should now
-	// successfully be written to update buffer to send back to CDS. This new
-	// update should contain the new child of LogicalDNS2.
-
-	select {
-	case chu := <-ch.updateChannel:
-		if diff := cmp.Diff(chu.updates, []xdsresource.ClusterUpdate{{
-			ClusterType: xdsresource.ClusterTypeEDS,
-			ClusterName: edsService,
-		}, {
-			ClusterType: xdsresource.ClusterTypeLogicalDNS,
-			ClusterName: logicalDNSService2,
-		}}); diff != "" {
-			t.Fatalf("got unexpected cluster update, diff (-got, +want): %v", diff)
-		}
-	case <-ctx.Done():
-		t.Fatal("Timed out waiting for the cluster update to be written to the update buffer.")
+	if err := compareLoadBalancingConfig(ctx, lbCfgCh, wantChildCfg); err != nil {
+		t.Fatal(err)
 	}
 }
 
-// TestUpdateRootClusterAggregateThenChangeRootToEDS tests the situation where
-// you have a fully updated aggregate cluster (where AggregateCluster success
-// test gets you) as the root cluster, then you update that root cluster to a
-// cluster of type EDS.
-func (s) TestUpdateRootClusterAggregateThenChangeRootToEDS(t *testing.T) {
-	// This initial code is the same as the test for the aggregate success case,
-	// except without validations. This will get this test to the point where it
-	// can update the root cluster to one of type EDS.
-	ch, fakeClient := setupTests()
-	ch.updateRootCluster(aggregateClusterService)
-
-	ctx, ctxCancel := context.WithTimeout(context.Background(), defaultTestTimeout)
-	defer ctxCancel()
-	_, err := fakeClient.WaitForWatchCluster(ctx)
-	if err != nil {
-		t.Fatalf("xdsClient.WatchCDS failed with error: %v", err)
-	}
-
-	fakeClient.InvokeWatchClusterCallback(xdsresource.ClusterUpdate{
-		ClusterType:             xdsresource.ClusterTypeAggregate,
-		ClusterName:             aggregateClusterService,
-		PrioritizedClusterNames: []string{edsService, logicalDNSService},
-	}, nil)
-	fakeClient.WaitForWatchCluster(ctx)
-	fakeClient.WaitForWatchCluster(ctx)
-	fakeClient.InvokeWatchClusterCallback(xdsresource.ClusterUpdate{
-		ClusterType: xdsresource.ClusterTypeEDS,
-		ClusterName: edsService,
-	}, nil)
-	fakeClient.InvokeWatchClusterCallback(xdsresource.ClusterUpdate{
-		ClusterType: xdsresource.ClusterTypeLogicalDNS,
-		ClusterName: logicalDNSService,
-	}, nil)
+// Tests the case where a requested cluster resource switches between being a
+// leaf and an aggregate cluster. In each of these cases, the test verifies that
```

Comment:
Same here about verbosity wrt aggregate cluster. Please specify what the aggregate cluster is pointing to.